pkg/app/minio.go

```diff
@@ -0,0 +1,143 @@
+package app
+
+import (
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
+)
+
+type minio struct {
+	ns string
+}
+
+var _ App = &minio{}
+
+func Minio(ns string) App {
+	return &minio{ns: ns}
+}
+
+func (a *minio) Name() string {
+	return "minio"
+}
+
+func (a *minio) Namespace() string {
+	return a.ns
+}
+
+func (a *minio) Install(t test.TestHelper) {
+	t.T().Helper()
+	if !oc.AnyResourceExist(t, "", "storageclass") {
+		t.Fatal("Your cluster doesn't contain any storageclass. Minio cannot be installed due to the required dynamic provisioning storage unavailable!")
+	}
+	oc.ApplyTemplate(t, a.ns, minioTemplate, nil)
+}
+
+func (a *minio) Uninstall(t test.TestHelper) {
+	t.T().Helper()
+	oc.DeleteFromTemplate(t, a.ns, minioTemplate, nil)
+}
+
+func (a *minio) WaitReady(t test.TestHelper) {
+	t.T().Helper()
+	oc.WaitDeploymentRolloutComplete(t, a.ns, "minio")
+}
+
+const minioTemplate = `
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+ # This name uniquely identifies the PVC. Will be used in deployment below.
+ name: minio-pv-claim
+ labels:
+   app: minio-storage-claim
+spec:
+ # Read more about access modes here: http://kubernetes.io/docs/user-guide/persistent-volumes/#access-modes
+ accessModes:
+   - ReadWriteOnce
+ resources:
+   # This is the request for storage. Should be available in the cluster.
+   requests:
+     storage: 1Gi
```

Comment:
Do you need 1Gi storage for testing ? I tested it with 256Mi and that is good enough.

Reply:
You are right, since it is not running during the whole test suite, only for one test